Sources/MeiliSearch/Error.swift

```diff
@@ -6,9 +6,54 @@ import Foundation
 public extension MeiliSearch {
 
   // MARK: Error
+  struct MSErrorResponse: Decodable, Equatable {
+    public let message: String
+    public let errorCode: String
+    public let errorType: String
+    public let errorLink: String?
+  }
+
+  static func errorHandler(url: URL, data: Data?, response: URLResponse?, error: Swift.Error?) throws {
+
+    // Communication Error with MeiliSearch
+    if let error: Swift.Error = error {
+      throw MeiliSearch.Error.meiliSearchCommunicationError(
+        message: error.localizedDescription,
+        url: url.absoluteString
+      )
+    }
+
+    guard let response = response as? HTTPURLResponse else {
+      fatalError("Correct handles invalid response, please create a custom error type")
+    }
+
+    // Error returned by MeiliSearch
+    if let unwrappedData: Data = data, let res: MeiliSearch.MSErrorResponse = try? Constants.customJSONDecoder.decode(MeiliSearch.MSErrorResponse.self, from: unwrappedData) {
+      throw MeiliSearch.Error.meiliSearchApiError(
+        message: res.message,
+        msErrorResponse: MeiliSearch.MSErrorResponse(
+          message: res.message,
+          errorCode: res.errorCode,
+          errorType: res.errorType,
+          errorLink: res.errorLink
+        ),
+        statusCode: response.statusCode,
+        url: url.absoluteString                            )
+    }
+
+    // HTTP error with MeiliSearch
+    if 400 ... 599 ~= response.statusCode {
+      throw MeiliSearch.Error.meiliSearchApiError(
+        message: HTTPURLResponse.localizedString(forStatusCode: response.statusCode),
+        msErrorResponse: nil,
+        statusCode: response.statusCode,
+        url: url.absoluteString
+      )
+    }
+  }
 
   /// Generic Error types for MeiliSearch,
-  enum Error: Swift.Error {
+  public enum Error: Swift.Error, LocalizedError, Equatable {
```

Comment:
```suggestion
  enum Error: Swift.Error, LocalizedError, Equatable {
```
Maybe you can remove the public here because you already are in:
`public extension MeiliSearch`